src/components/CentralHeader/CentralHeader.tsx

```diff
@@ -31,22 +32,42 @@ interface ICentralHeaderProps extends RouteComponentProps {
 /**
  * Central header right below Slim header, containing app title and user name and user role with dropdown to go to profile page and logout
  */
-export const CentralHeader = withRouter<
-  ICentralHeaderProps,
-  React.FC<ICentralHeaderProps>
->((props: ICentralHeaderProps) => {
+export const CentralHeader = withRouter((props: ICentralHeaderProps) => {
   /**
    * react-i18next translation hook
    */
   const { t } = useTranslation();
 
+  /**
+   * add user icon if user name is set
+   */
+  const userIcon =
+    props.userName.trim() !== "" ? (
+      <svg className="icon icon-white">
+        <use xlinkHref={`${bootstrapItaliaImages}#it-user`} />
+      </svg>
+    ) : null;
+
+  /**
+   * switch from user role enum to string
+   */
+  const userRole = (userRoleEnum => {
+    switch (userRoleEnum) {
+      case UserRoleEnum.ORG_DELEGATE:
+        return t("centralHeader.userBadge.delegate");
+      case UserRoleEnum.DEVELOPER:
+        return t("centralHeader.userBadge.developer");
+      case UserRoleEnum.ORG_MANAGER:
+        return t("centralHeader.userBadge.legalRep");
+      case UserRoleEnum.ADMIN:
+        return t("centralHeader.userBadge.admin");
+    }
+  })(props.userRole);
   /**
    * create badge for user role if user role is set
    */
   const userRoleBadge =
-    props.userRole !== "" ? (
-      <Badge color="white">{props.userRole}</Badge>
-    ) : null;
+    props.userRole !== "" ? <Badge color="white">{userRole}</Badge> : null;
```

Comment:
consider to make a component `<UserRoleBadge >` that takes userRole as input prop (we also have a NonEmptyString type if you dare : )

Reply:
I see the advantage of creating child components for a more ordered code and to give each component it's logic, but in this case the amount of code is quite small...which would be the advantage of creating a component for such few lines of code?
about NonEmptyString class, I only find NonEmptyArray in fp-ts...where can I find it?